src/main/java/seedu/address/model/person/RoleContainsKeywordsPredicate.java

```diff
@@ -1,32 +1,55 @@
 package seedu.address.model.person;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.function.Predicate;
 
+import seedu.address.commons.util.ToStringBuilder;
+
 /**
- * Represents a predicate that checks if a person's role contains a specified keyword.
+ * Represents a predicate that checks if a person's role contains any of the specified keywords.
  * This predicate is used to filter a list of persons based on their roles.
  */
 public class RoleContainsKeywordsPredicate implements Predicate<Person> {
-    private final String keyword;
+    private final List<String> keywords;
 
     /**
-     * Constructs a {@code RoleContainsKeywordsPredicate} with the specified keyword.
+     * Constructs a {@code RoleContainsKeywordsPredicate} with a single keyword.
      *
      * @param keyword The keyword to match against the person's role.
      */
     public RoleContainsKeywordsPredicate(String keyword) {
-        this.keyword = keyword.toLowerCase();
+        this.keywords = new ArrayList<>();
+        this.keywords.add(keyword.toLowerCase());
```

Comment:
Can consider using `Collections#singletonList(T)`. Safer this way too since it is a immutable list.